src/overlays/actors/ovl_Obj_Tsubo/z_obj_tsubo.c

```diff
@@ -46,39 +73,275 @@ static ColliderCylinderInit D_80BA1B94 = {
+    this->actor.room = -1;
+    func_8002F7DC(&this->actor, NA_SE_PL_PULL_UP_POT);
+    this->actor.flags |= 0x10;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Tsubo/func_80BA180C.s")
+void ObjTsubo_LiftedUp(ObjTsubo* this, GlobalContext* globalCtx) {
+    if (Actor_HasNoParent(&this->actor, globalCtx) != 0) {
+        this->actor.room = globalCtx->roomCtx.curRoom.num;
+        func_80BA188C(this);
+        ObjTsubo_Fall(this);
+        func_8002D7EC(&this->actor);
+        Actor_UpdateBgCheckInfo(globalCtx, &this->actor, 5.0f, 15.0f, 0.0f, 0x85);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Tsubo/func_80BA188C.s")
+void func_80BA188C(ObjTsubo* this) {
+    this->actor.velocity.x = Math_SinS(this->actor.world.rot.y) * this->actor.speedXZ;
+    this->actor.velocity.z = Math_CosS(this->actor.world.rot.y) * this->actor.speedXZ;
+    this->actor.colChkInfo.mass = 0xF0;
+    D_80BA1B50 = (Rand_ZeroOne() - 0.7f) * 2800.0f;
+    D_80BA1B58 = (Rand_ZeroOne() - 0.5f) * 2000.0f;
+    D_80BA1B54 = 0;
+    D_80BA1B5C = 0;
+    this->actionFunc = func_80BA1958;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Tsubo/func_80BA1958.s")
+void func_80BA1958(ObjTsubo* this, GlobalContext* globalCtx) {
+    s32 pad[2];
+    if ((this->actor.bgCheckFlags & 0xB) != 0 || (this->collider.base.atFlags & 2) != 0) {
+        ObjTsubo_AirBreak(this, globalCtx);
+        ObjTsubo_SpawnCollectible(this, globalCtx);
+        Audio_PlaySoundAtPosition(globalCtx, &this->actor.world.pos, 0x14, 0x2887U);
+        Actor_Kill(&this->actor);
+    } else if ((this->actor.bgCheckFlags & 0x40) != 0) {
+        ObjTsubo_WaterBreak(this, globalCtx);
+        ObjTsubo_SpawnCollectible(this, globalCtx);
+        Audio_PlaySoundAtPosition(globalCtx, &this->actor.world.pos, 0x14, 0x2887U);
+        Actor_Kill(&this->actor);
+    } else {
+        ObjTsubo_Fall(this);
+        func_8002D7EC(&this->actor);
+        Math_StepToS(&D_80BA1B54, D_80BA1B50, 0x64);
+        Math_StepToS(&D_80BA1B5C, D_80BA1B58, 0x64);
+        this->actor.shape.rot.x += D_80BA1B54;
+        this->actor.shape.rot.y += D_80BA1B5C;
+        Actor_UpdateBgCheckInfo(globalCtx, &this->actor, 5.0f, 15.0f, 0.0f, 0x85);
+        Collider_UpdateCylinder(&this->actor, &this->collider);
+        CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+        CollisionCheck_SetAT(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Tsubo/ObjTsubo_Update.s")
+void ObjTsubo_Update(Actor* thisx, GlobalContext* globalCtx) {
+    ObjTsubo* this = THIS;
+    this->actionFunc(this, globalCtx);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Tsubo/func_80BA1B0C.s")
+void ObjTsubo_Draw(ObjTsubo* this, GlobalContext* globalCtx) {
+    Gfx_DrawDListOpa(globalCtx, D_80BA1B84[(this->actor.params >> 8) & 1]);
+}
```

Comment:
make sure to run the formatter (it will add a newline at the end of the file for you as well)